browser_use/agent/service.py

```diff
@@ -1197,10 +1194,14 @@ def save_history(self, file_path: str | Path | None = None) -> None:
 			file_path = 'AgentHistory.json'
 		self.state.history.save_to_file(file_path)
 
+	async def wait_until_resumed(self):
+		await self._external_pause_event.wait()
+
 	def pause(self) -> None:
 		"""Pause the agent before the next step"""
 		print('\n\n⏸️  Got Ctrl+C, paused the agent and left the browser open.')
 		self.state.paused = True
```

Comment:
do we even need `self.state.paused` anymore if we are using `self._external_pause_event` as the semaphore? I think we should remove it if it's not needed